src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -978,7 +991,17 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
                             }
 
                             utf16Data = Vector128.CreateScalarUnsafe(possibleNonAsciiQWord).AsInt16();
-                            Unsafe.WriteUnaligned<uint>(pOutputBuffer, Sse2.ConvertToUInt32(Sse2.PackUnsignedSaturate(utf16Data, utf16Data).AsUInt32()));
+
+                            if (AdvSimd.IsSupported)
+                            {
+                                Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
+                                Vector128<byte> source = AdvSimd.ExtractNarrowingSaturateUnsignedUpper(lower, AdvSimd.LoadVector128((short*)pInputBuffer + Vector128<short>.Count));
+                                AdvSimd.StoreSelectedScalar((uint*)pOutputBuffer, source.AsUInt32(), 0);
+                            }
+                            else
+                            {
+                                Unsafe.WriteUnaligned<uint>(pOutputBuffer, Sse2.ConvertToUInt32(Sse2.PackUnsignedSaturate(utf16Data, utf16Data).AsUInt32()));
```

Comment:
I wonder why this isn't using `Sse2.StoreScalar`...

Reply:
Was this approach wrong for the AdvSimd case?

Reply:
I took another look and it may have been incorrect, but I'd like @echesakovMSFT to confirm.

Reply:
I think Tanner just pointed here that using `Sse2.StoreScalar` would be more preferable on x64 here since the code could be simplified to

```c#
Sse2.StoreScalar(pOutputBuffer, Sse2.PackUnsignedSaturate(utf16Data, utf16Data).AsUInt32());
```
But I am not sure what happens when pOutputBuffer is unaligned. @tannergooding comments?

Reply:
@tannergooding This code was introduced in .NET Core 3.0. `Sse2.StoreScalar` wasn't introduced until .NET 5.0. See https://github.com/dotnet/runtime/pull/34085.